pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -68,6 +70,18 @@ func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
 }
 
 func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry, raw json.RawMessage, apiVersion string, resourceID string) api.ValidationResult {
+	log.Infof("running preflight validation on resource: %s", resourceID)
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	isCreate := cosmosdb.IsErrorStatusCode(err, http.StatusNotFound)
+	if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusNotFound) {
+		return api.ValidationResult{
```

Comment:
Why is there a new api.ValidationResult type instead of using the existing `pkg/api/error.go` -> `CloudError` struct?

Reply:
The documentation described the response for preflight to look like:
`type response = {
  status: "Failed" | "Succeeded";
  error: error;
}

type error = {
  code: string;
  message: string;
  target: string;
  details: error[];
}`

I saw that clouderror has a status code field, I didn't want that potentially interfering. The error looks to be the same as clouderrorbody so I could use that instead of making a new one.